common/analysis/linter_test_utils.h

```diff
@@ -72,10 +72,27 @@ void RunLintTestCase(const LintTestCase& test,
   // Instantiate a linter that runs a single rule to analyze text.
   LintRunner<RuleType> lint_runner(make_rule());
   const LintRuleStatus rule_status = lint_runner.Run(analyzer.Data(), filename);
-  const auto& violations(rule_status.violations);
```

Comment:
I started modifying the `LintRuleStatus` class to use a `set` instead of the `vector` to store the violations.

In general it works quite well and the code needed to solve what this PR is trying to solve is indeed much prettier, but there is one issue: It looks as though the tree can be created without any TokenInfos. This makes sorting by the text pointers useless - moreover having a set there doesn't really work, as with  a `operator<` definition that is based on the token info text pointers the items are not inserted correctly.

As far as I can see, such trees are only created in the tests, and some tests fail for that reason, for example: https://github.com/google/verible/blob/37eec341f5390744aeee70fe91417c19ceb8f10b/common/analysis/syntax_tree_linter_test.cc#L243

How should this be approached?
* Should I assume that this API will only ever be used to write tests and therefore assume it is safe to modify those tests?
* Should I keep collecting the violations as a vector but add something like `getViolationsAsSet`, and use it for linter tests + printing linter violations?

Do you have any other suggestions?

Reply:
Ah yes, those synthetic trees in those tests will need to be populated with slightly more real leaf tokens.

Here are some examples where I needed tokens to point to `substr` ranges inside a contiguous buffer (string_view):

https://github.com/google/verible/blob/81d39b1e13688850fb918cef9b5c47f135a962dd/common/text/tree_utils_test.cc#L250

https://github.com/google/verible/blob/81d39b1e13688850fb918cef9b5c47f135a962dd/common/text/text_structure_test.cc#L434

You may certainly update those tests.

Reply:
Thanks! Updated those tests based on your suggestion.